test/e2e/src/core/e2e_tests.sh

```diff
@@ -77,15 +77,25 @@ validateCommonParameters() {
 	   echo "ERROR: parameter TENANT_ID is required." > ${results_dir}/error
 	   python3 setup_failure_handler.py
 	fi
-	if [ -z $CLIENT_ID ]; then
-	   echo "ERROR: parameter CLIENT_ID is required." > ${results_dir}/error
-	   python3 setup_failure_handler.py
-	fi
+   ## Look for WORKLOAD_CLIENT_ID
+   if [[ -z "${WORKLOAD_CLIENT_ID}" ]]
+   then
+      if [ -z $CLIENT_ID ]; then
+         echo "ERROR: parameter CLIENT_ID is required." > ${results_dir}/error
+         python3 setup_failure_handler.py
+      fi
 
-	if [ -z $CLIENT_SECRET ]; then
-	   echo "ERROR: parameter CLIENT_SECRET is required." > ${results_dir}/error
-	   python3 setup_failure_handler.py
-	fi
+      if [ -z $CLIENT_SECRET ]; then
+         echo "ERROR: parameter CLIENT_SECRET is required." > ${results_dir}/error
+         python3 setup_failure_handler.py
+      fi
+   else
+    if [[ -z "${CLIENT_ID}" ]]
```

Comment:
this else clause is for when WORKLOAD_CLIENT_ID exists... why are we checking for CLIENT_ID's absence in this case?

Reply:
> this else clause is for when WORKLOAD_CLIENT_ID exists... why are we checking for CLIENT_ID's absence in this case?

good catch. fixed.